MidCamp/midcamp
# 3252663: Remove disused modules, disable config_installer

## Issue

[#3252663 Remove uninstalled modules from the project](https://www.drupal.org/project/midcamp/issues/3252663)

## Description 

- Removes Geocoder, Geofield, Geolocation, and Media Entity modules (which were not installed)
- Updates the Entity API module for D9 compatibility 
- Patches Entity Reference Revisions module for Composer warnings thrown when executing Drush commands
- Deprecates config_installer, which is incompatible with D9.  This work includes:
  - Updates to `core.extension.yml` replacing `config_installer` with `minimal`
  - `settings.php` updates removing references to `config_installer`
  - Update hook in `midcamp_utility.install` to install the `minimal` profile
  - Updates to Lagoon deployment post-rollout tasks in `.lagoon.yml` to run the database update + config import sequence twice, with the first round modified to include `|| true` (which will allow the script to continue if a failure is encountered...see [deployment log lagoon-build-9dx4a8](https://dashboard.amazeeio.cloud/projects/midcamp-org/midcamp-org-feature-3252663-remove-disused-modules/deployments/lagoon-build-9dx4a8) to see the failure this resolves).  This is necessary to circumvent a race condition whereby the first attempt at each of these updates will fail like:
    ```shell
    The following updates are pending:

    midcamp_utility module : 
      8901 -   Switch from 'config_installer' profile to 'minimal'. 

    Do you wish to run all pending updates? (y/n): y
    Configuration objects provided by <em                                    [error]
    class="placeholder">minimal</em> have unmet dependencies: <em
    class="placeholder">block.block.stark_admin (stark)</em>
    Performing midcamp_utility_update_8901                                      [ok]
    Skipping cache-clear operation due to --cache-clear=0 option.          [warning]
    Failed: Configuration objects provided by minimal have unmet             [error]
    dependencies: block.block.stark_admin (stark)
    Skipping cache-clear operation due to --cache-clear=0 option.               [ok]
    Finished performing updates.                                                [ok]
    command terminated with exit code 1
    ```
  - For consistency with `.lagoon.yml` changes made above, the same sequence has been updated in `.amazeeio.yml` 

## Test instructions

- Install updated dependencies via `lando composer install`
- Perform database updates and configuration imports twice, matching the updates made to the `.lagoon.yml` file:
  - Run `lando drush updb -y`
  - Run `lando drush cim -y`
  - Run `lando drush updb -y` again
  - Run `lando drush cim -y` again
- Navigate to http://midcamp-org.lndo.site/admin/reports/upgrade-status and observe the following:
  - Geocoder, Geofield, Geolocation, and Media Entity modules are no longer listed
  - Entity module is listed as compatible
- Navigate to http://midcamp-org.lndo.site/admin/reports/status and confirm the installation profile is listed as "minimal"
- Review the remote environment associated with this branch at https://nginx.feature-3252663-remove-disused-modules.midcamp-org.us2.amazee.io/ and confirm the items above are reflected there as well.
